### DISCUSSION TITLE
Uploads over 1MB seem to fail.

### DISCUSSION THREAD

**Discussion by nwhobart** [open-webui/open-webui#6566] [Issues]
I have a pretty stock version of open-webui running on our Kubernetes cluster. When I try to upload a file to be analyzed that's over 1MB (give or take) the spinner just spins and nothing happens. In this example below I'm trying to upload the  ToS for macOS Sequoia. I can leave this sitting forever and nothing will happen. Logs for the pod aren't showing anything either which adds to the frustration. If anyone has run into this I'd appreciate any help.

I am running `open-webui:0.3.35`.

![image](https://github.com/user-attachments/assets/fd1de733-3c3e-4465-98ba-bf7f09ef40ae)

**Comment by peuportier** [comment]
set the size maybe give value in document 
<img width="1184" alt="Screenshot 2024-10-30 at 16 00 22" src="https://github.com/user-attachments/assets/b0011b61-1c29-4727-9eed-ef5a6c35ce72">
 give it maybe 50mb

<img width="799" alt="Screenshot 2024-10-30 at 16 03 09" src="https://github.com/user-attachments/assets/d53b0d6e-a79f-4edf-84f4-215993b10472">

**Comment by nwhobart** [comment]
I set the size to `100` in the console but it didn't make any difference.

**Comment by dlowings** [comment]
I have the same issue trying to upload a single 4.5MB pdf. Is this a new issue, the wheel just spins and never uploads. eventually it fails BUT the full complete document shows up in the upload directory . If you re upload it, you will get a second copy so on and so forth but it never ends up in the workspace.
<img width="676" alt="Screen Shot 2024-10-30 at 8 05 58 PM" src="https://github.com/user-attachments/assets/70cb0e98-04c7-49e4-955d-300af6ddd91c">
<img width="892" alt="Screen Shot 2024-10-30 at 8 08 05 PM" src="https://github.com/user-attachments/assets/274203bc-c396-4207-99b0-931e6f285f85">

**Reply by dlowings** [reply]
For what its worth, I reinstalled Open-WebUI after doing a complete removal and after that it is now working.

**Comment by peuportier** [comment]
Interesting. What system are you using too for install OWUI? MAC / LINUX or WINDOWS? looks like the permission not correct. if linux maybe on the files to give a 755 rights would help. chmod -R 755 / files for your uploads . But we need further information for check if related to OWUI in your environment . I use MAC os , I don't have this issue with last version of OWUI. Please let me know

**Reply by peuportier** [reply]
Also change embedding model. maybe the embedding model doesn't do his job properly

**Reply by nwhobart** [reply]
We're using the Kubrernetes Helm chart

**Reply by nwhobart** [reply]
Changing the embedding model didn't work. Also I doubt it's a permissions issue as file sizes under 1MB upload just fine.

**Comment by filiptrplan** [comment]
I'm having the same issue. Works for files at about 100kb but anything over 1MB just spins indefinitely.

**Reply by beastech** [reply]
Are you using Mac, Windows, or Linux to run OWUI?

**Reply by filiptrplan** [reply]
I'm using Linux.

**Reply by filiptrplan** [reply]
It seems it was a problem with the embedding model. After using the local one it seems to work. Then I changed the batch size on the OpenAI model to 500 and it also worked after some time. Maybe the documentation should specify that it could fail for large files if the batch size is too small?

The embeddings are really slow though. LobeChat doesn't seem to suffer from the same problem, taking just a couple of seconds for each file even if 100s of pages long.

**Reply by nwhobart** [reply]
My chunk size is 1000. Still no luck.

![image](https://github.com/user-attachments/assets/013a5d99-d908-44e5-88fe-d1c41230e14c)

**Reply by nwhobart** [reply]
![image](https://github.com/user-attachments/assets/03572d51-3148-43a6-bfc5-edcb2d51f625)

**Reply by filiptrplan** [reply]
I maxed out the batch size and it is much faster now. Not sure what the problem on your side is. I also changed my vector DB to the other one (forgot the name).

**Comment by M0ritzFeickert** [comment]
I had the same issue, everything above 1MB would not work. For me, it was my nginx reverse proxy not allowing requests bigger than 1MB. If you are also using nginx then set the `client_max_body_size` to more than 1MB

**Reply by nwhobart** [reply]
Did you make that change on the ingress for openweb or at the controller level?

**Reply by nwhobart** [reply]
For future folks, adding: 

```
controller:
  config:
     proxy-body-size: "35m"
```

to my `values.yaml` file was the key for me.

**Reply by jormeijer** [reply]
> For future folks, adding:
> 
> ```
> controller:
>   config:
>      proxy-body-size: "35m"
> ```
> 
> to my `values.yaml` file was the key for me.

You can also use Ingress annotations:

```
ingress:
  annotations:
    nginx.ingress.kubernetes.io/proxy-body-size: 128m
```

**Comment by DJJones66** [comment]
I just did a quick test on Windows - Docker install and it worked as I would have expected in the interface and nothing jumped out in the logs.
